File: javascript/349.IntersectionofTwoArrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    var nums=[];
    nums2.filter(function(elem, pos) {
        if(nums2.indexOf(elem) === pos&&nums1.indexOf(elem)!=-1 ){
            nums.push(nums2[pos]);
        }
    }); 

    return nums;
};

File: python/349.IntersectionofTwoArrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        table={}
        result = []
        for n in nums1:
            table[n]=1
        for n in nums2:
            if n in table: 
                if table[n]==1:
                    result.append(n)
                    table[n]=2
        return result
